client/src/routes/register/+page.svelte

```diff
@@ -1,47 +1,48 @@
 <script lang="ts">
-  import { z } from 'zod';
-  import { superForm, setMessage } from "sveltekit-superforms/client";
-    import { goto } from '$app/navigation';
+  import { z } from "zod";
+  import {
+    superForm,
+    superValidateSync,
+    setError,
+  } from "sveltekit-superforms/client";
+  import { goto } from "$app/navigation";
+  import type { ErrorResponse } from "bindings/ErrorResponse";
 
-  let form_data;
-
-  const userSchema = z.object({
+  const registerSchema = z.object({
     username: z.string().min(5),
     email: z.string().email(),
     password: z.string().min(8),
   });
 
-  const { form, errors, message, constraints, enhance } = superForm(
-    form_data,
+  const { form, errors, constraints, enhance } = superForm(
+    superValidateSync(registerSchema),
     {
       SPA: true,
-      validators: userSchema,
+      validators: registerSchema,
       onUpdate: async ({ form }) => {
-        console.log(form.valid);
         if (form.valid) {
-          setMessage(form, "Valid data!");
           const res = await fetch("http://localhost:6060/api/v1/users", {
             method: "POST",
             headers: {
               "Content-Type": "application/json",
             },
-            body: JSON.stringify(form.data)
+            body: JSON.stringify(form.data),
           });
-
-          await goto("/login");
+          if (res.status >= 400) {
+            const err: ErrorResponse = await res.json();
+            const errField = err.error.includes("email") ? "email" : "username";
```

Comment:
this doesn't include the errors that could happen in the password field, only email and username

Reply:
the only constrain is that the length being more than 8 and that's covered by `registerSchema` no?

Reply:
fair enough